pkg/storage/bucket/azure/config.go

```diff
@@ -30,6 +31,7 @@ func (cfg *Config) RegisterFlags(f *flag.FlagSet) {
 func (cfg *Config) RegisterFlagsWithPrefix(prefix string, f *flag.FlagSet) {
 	f.StringVar(&cfg.StorageAccountName, prefix+"azure.account-name", "", "Azure storage account name")
 	f.Var(&cfg.StorageAccountKey, prefix+"azure.account-key", "Azure storage account key. If unset, Azure managed identities will be used for authentication instead.")
+	f.Var(&cfg.StorageConnectionString, prefix+"azure.connection-string", "If `connection-string` is set, the values of `account-name` and `endpoint-suffix` values will not be used. Use this method over `account-key` if you need to authenticate via a SAS token. Or if you use the Azurite emulator.")
```

Comment:
> the values of `account-name` and `endpoint-suffix` values will not be used

I realize this derives from the usage string from objstore, but it doesn't seem true. The account name is used in the [validate](https://github.com/thanos-io/objstore/blob/ff7faac741fba6758e99adf1bbce618e25147af6/providers/azure/azure.go#L88) function for the azure configuration and it will error if it isn't provided.